core/src/util/usLog.h

```diff
@@ -0,0 +1,101 @@
+/*=============================================================================
+
+  Library: CppMicroServices
+
+  Copyright (c) The CppMicroServices developers. See the COPYRIGHT
+  file at the top-level directory of this distribution and at
+  https://github.com/saschazelzer/CppMicroServices/COPYRIGHT .
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+=============================================================================*/
+
+#ifndef USLOG_P_H
+#define USLOG_P_H
+
+#include <usCoreConfig.h>
+
+#include <iostream>
+#include <sstream>
+#include <memory>
+#include <mutex>
+
+namespace us {
+
+class LogSink : public std::enable_shared_from_this<LogSink> 
+{
+public:
+  explicit LogSink(std::ostream* sink, bool enable = false) : _enable(enable), _sink_lock(), _sink(sink) 
+  {
+    if (_sink == nullptr) _enable = false;
+  }
+
+  LogSink() = delete;
+  LogSink(const LogSink&) = delete;
+  LogSink& operator=(const LogSink&) = delete;
+  ~LogSink() = default;
+
+  bool Enabled() { return _enable; }
+
+  void Log(const std::string& msg)
+  {
+    if (!_enable) return;
+    std::lock_guard<std::mutex> lock(_sink_lock);
+    *_sink << msg;
+  }
+
+private:
+  bool _enable;
+  std::mutex _sink_lock;
```

Comment:
I suggest to inherit the `LogSink` class from `MultiThreaded<>` and not use a `std::mutex` directly. Then this code will work properly in the single-threaded CMake configuration as well.